src/Views/PartitioningView.vala

```diff
@@ -0,0 +1,158 @@
+// -*- Mode: vala; indent-tabs-mode: nil; tab-width: 4 -*-
+/*-
+ * Copyright (c) 2016-2018 elementary LLC. (https://elementary.io)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authored by: Michael Aaron Murphy <michael@system76.com>
+ */
+
+public class Installer.PartitioningView : AbstractInstallerView  {
+    public signal void next_step ();
+
+    private Gtk.Button next_button;
+    private Distinst.Disks disks;
+    private GLib.Array<PartitionBar> partitions;
+
+    public GLib.Array<Installer.Mount> mounts;
+
+    public static uint64 minimum_disk_size;
+
+    public PartitioningView (uint64 size) {
+        minimum_disk_size = size;
+        Object (cancellable: true);
+    }
+
+    construct {
+        this.mounts = new GLib.Array<Installer.Mount> ();
+        this.margin = 24;
+        var disk_list = new Gtk.Grid ();
+        disk_list.row_spacing = 24;
+        var disk_scroller = new Gtk.ScrolledWindow (null, null);
+        disk_scroller.expand = true;
+        disk_scroller.vscrollbar_policy = Gtk.PolicyType.NEVER;
+        disk_scroller.add (disk_list);
+
+        var description = new Gtk.Label (_("Select which partitions to use across all drives. This will erase all data on the selected partitions."));
+
+        this.content_area.attach(description, 0, 0, 1, 1);
```

Comment:
It's unnecessary to use `this`.
Space before ()
Default width and height for grid is `1` so you can omit the last two args here